### ISSUE TITLE
As a Manifest creator, blogger, website author, I want to be able to easily link different viewers to open a IIIF resource

### ISSUE THREAD

**Issue by klokan** [IIIF/iiif-stories#85]
Link to open a document in a viewer

# Description

Having a link to a IIIF resource (manifest json) we should present on a website one or more links which opens this resource in a viewer like Mirador (probably a public demo instance) or in various other viewers or tools operated by external parties.

The link format should be ideally standardised / recommended or specified as best practice and used across different viewers.

Drag&drop is great - and required for adding a second or third IIIF resource into a viewer - but the first image / document should be better openable by simply clicking on a link which loads the viewer with this document in a new window! It is easier for people to click, instead of doing a drag&drop.

To construct such link is now not possible (at least not to the public Mirador demo instance), and format of such link is not universal across viewers and tools.

It would be good to define best practice in this area - and make implementation in most popular viewers (Mirador, UniversalViewer, Georeferencer, etc.)

## Proposed Solutions

```http://projectmirador.org/demo/#URL```

but it can be probably improved. A discussion is welcome.

## Additional Background

More details and a suggestion for implementation inside of Mirador is under
https://github.com/ProjectMirador/mirador/issues/1246

**Comment by klokan** [comment]
I don't have rights to add labels on the issue. Could you, please add "discovery" and "discovery-import" @zimeon or @aisaac here?

**Comment by aisaac** [comment]
here you go, Peter :-)

**Comment by klokan** [comment]
Thanks @aisaac

**Comment by zimeon** [comment]
Behavior of `ViewerInstanceURL#URL` is expected to be same as pasting or dragging-and-dropping `URL` into empty `ViewerInstanceURL`?

I wonder whether there is a problem of conflict with other uses of `#whatever` by viewers?

**Comment by klokan** [comment]
> Behavior of ViewerInstanceURL#URL is expected to be same as pasting or dragging-and-dropping URL into empty ViewerInstanceURL?

Yes. But it is a basic click on a hyperlink - like any other link on the web.
Very natural and easiest possible for any online visitor.

> I wonder whether there is a problem of conflict with other uses of #whatever by viewers?

There can be - especially if the viewers have permalink (such as position on the image, status of the tools or controls in the viewer, etc.).

Still - the viewers can react on this predefined `#URL` link always - and clear it with JavaScript as soon as they are initialized and possibly replace it with their native permalink format behind `#`, if required. They can offer their own format of "advanced permalink", which is viewer specific  - but still support the "universal simple permalink" just with `#URL`

The regex for testing the `#URL` would be probably `#http.*` - so startswith('http'), unless there are other protocols over which the iiif can be served (file:// or another?)

**Comment by azaroth42** [comment]
Discussion on call:

Difficult to standardize a link that opens with all of the functionality that is desired from other use cases.  The particular viewer instance creation needs to be out of band -- any template could be also out of band.  Very dependent on the clients.  Seems like a solution for import in general -- the requirement being instantiate / import a manifest to  a client with a click rather than drag and drop, the solution of a URL pattern is one solution, but not necessarily the right approach.

Would still be good to see convergence in this space, but not a priority until we can implement import in general.   Client coordination is good. 

If it is clicking to get to different viewers then that's similar to how we all implement social sharing buttons our our sites.
